### DISCUSSION TITLE
Parameterised searchable() and sortable() Table methods

### DISCUSSION THREAD

**Discussion by Blackpig** [filamentphp/filament#51] [Ideas]
Hey guys,

Nice work so far, I've been playing around and migrating a small app to see how easy/mature things are.

It would be nice to be able to pass in column names into the searchable/sortable methods of the Table column.  If I access an attribute on the Eloquent model e.g. `getFullNameAttribute` -> which returns the `first_name . '  ' . last_name` as concatenated string then the searchable/sortable methods fail because there is no underlying `full_name` column on the underlying table. 

So I was thinking something like the below to pass in the query builder

```
Columns\Text::make('full_name')
 ->primary()
->searchable(['last_name', 'first_name'])
->sortable(['last_name', 'first_name']),
```

Edit:
Obviously as work around for the searchable it's possible to add the individual fields/columns as hidden 

Keep up the good work

**Comment by danharrin** [comment]
This is perfect! I've considered this issue before but hadn't thought of a suitable solution. Will have this in the core very soon. Thanks!

**Comment by ManojKiranA** [comment]
This is possible in v3
